src/main/io/ledstrip.c

```diff
@@ -496,6 +497,31 @@ static const struct {
 
 static void applyLedFixedLayers(void)
 {
+    // For loading bar style layers:
+#ifdef USE_GPS
+    int totalLedsGps = 0;
+    int lightsCountGps = 0;
+#endif
+    int totalLedsBattery = 0;
+    int lightsCountBattery = 0;
+
+    for (int ledIndex = 0; ledIndex < ledCounts.count; ledIndex++) {
```

Comment:
I think calculating the total leds in a bar only needs to be updated when the LED strip config changes, see `reevaluateLedConfig`  perhaps move it to a new function called `updateLedBars` and call it from `updateRequiredOverlay`.

Maybe also move the `*Gps` and `*Battery` counters to a structure, something like `ledBarState` which contains `total` and `count`, then define something like `barStates` and an enum.

```
typedef struct ledBarState_s {
    u_int8 total;
    u_int8 count;
} barState_t;

#define LED_BAR_GPS 0
#define LED_BAR_BATTERY 1
#define MAX_BAR_STATES 2

ledBarState_t ledBarStates[MAX_BAR_STATES] = 0;

...

ledBarState[LED_BAR_GPS].total++;
```

Reply:
Also, bar color is recomputed for every led (and discarded nor non-active leds in bar).

Reply:
Thanks, I addressed this in a new commit.

Reply:
great stuff, another benefit is that it's easier to debug too just by looking at the `ledBarStates` variable.